implementations/rust/ockam/ockam_abac/src/policy.rs

```diff
@@ -26,12 +26,11 @@ pub struct PolicyAccessControl {
 /// Debug implementation writing out the resource, action and initial environment
 impl Debug for PolicyAccessControl {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let resource = &self.resource;
-        let action = &self.action;
-        let environment = &self.environment;
-        f.write_str(format!("resource {resource:?}").as_str())?;
-        f.write_str(format!("action {action:?}").as_str())?;
-        f.write_str(format!("environment {environment:?}").as_str())
+        f.debug_struct("PolicyAccessControl")
+         .field("resource", &self.resource)
+         .field("action", &self.action)
+         .field("environment", &self.environment)
```

Comment:
reviewing myself xD

shouldn't these be like `.field("addr", &format_args!("{}", self.addr))` ? (cf. [docs](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct))

Reply:
Isn't it exactly the same?